packages/babel-parser/src/plugins/flow.js

```diff
@@ -2830,4 +2884,419 @@ export default (superClass: Class<Parser>): Class<Parser> =>
+        if (memberName === "") {
+          continue;
+        }
+        if (/^[a-z]/.test(memberName)) {
+          this.flowEnumErrorInvalidMemberName(id.start, {
+            enumName,
+            memberName,
+          });
+        }
+        if (seenNames.has(memberName)) {
+          this.flowEnumErrorDuplicateMemberName(id.start, {
+            enumName,
+            memberName,
+          });
+        }
+        seenNames.add(memberName);
+        const context = { enumName, explicitType, memberName };
+        memberNode.id = id;
+        switch (init.type) {
+          case "boolean": {
+            this.flowEnumCheckExplicitTypeMismatch(
+              init.pos,
+              context,
+              "boolean",
+            );
+            memberNode.init = init.value;
+            members.booleanMembers.push(
+              this.finishNode(memberNode, "EnumBooleanMember"),
+            );
+            break;
+          }
+          case "number": {
+            this.flowEnumCheckExplicitTypeMismatch(init.pos, context, "number");
+            memberNode.init = init.value;
+            members.numberMembers.push(
+              this.finishNode(memberNode, "EnumNumberMember"),
+            );
+            break;
+          }
+          case "string": {
+            this.flowEnumCheckExplicitTypeMismatch(init.pos, context, "string");
+            memberNode.init = init.value;
+            members.stringMembers.push(
+              this.finishNode(memberNode, "EnumStringMember"),
+            );
+            break;
+          }
+          case "invalid": {
+            this.flowEnumErrorInvalidMemberInitializer(init.pos, context);
+            break;
+          }
+          case "none": {
+            switch (explicitType) {
+              case "boolean":
+                this.flowEnumErrorBooleanMemberNotInitialized(
+                  init.pos,
+                  context,
+                );
+                break;
+              case "number":
+                this.flowEnumErrorNumberMemberNotInitialized(init.pos, context);
+                break;
+              default:
+                members.defaultedMembers.push(
+                  this.finishNode(memberNode, "EnumDefaultedMember"),
+                );
+            }
+          }
+        }
+
+        if (!this.match(tt.braceR)) {
+          this.expect(tt.comma);
+        }
+      }
+      return members;
+    }
+
+    flowEnumStringBody(
+      bodyNode: N.Node,
+      initializedMembers: Array<N.Node>,
+      defaultedMembers: Array<N.Node>,
+      { enumName }: { enumName: string },
+    ): N.Node {
+      if (initializedMembers.length === 0) {
+        bodyNode.members = defaultedMembers;
+      } else if (defaultedMembers.length === 0) {
+        bodyNode.members = initializedMembers;
+      } else if (defaultedMembers.length > initializedMembers.length) {
+        bodyNode.members = defaultedMembers;
+        for (const member of initializedMembers) {
```

Comment:
Why do we need both the cases? (`defaultMembers` and `initializedMembers`)

Reply:
Basically, we either allow all enum members to be initialized, or not initialized. It has to be consistent. This is unlike TypeScript (for example), so users will probably try to mix up initializing and not-initializing (defaulting). That's why we have extra code and logic to come up with good errors, as it is likely for users to experiment and try it out. We track both initialized and defaulted members, and report errors on whatever is the minority. For example:

```
enum E {
  A,
  B = 2,
  C = 3,
}
```
A naive implementation would take the first member as correct (using a defaulted member), and error on the other two (`B` and `C`), as they are initialized. We are smarter: we see that the majority of members are initialized, and instead error on the minority - member `A`.